dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -405,72 +408,71 @@ private void handleDropEvent(final DragEvent event, final TreeTableRow<Ds3TreeTa
-                LOG.info("Passing new Ds3PutJob to jobWorkers thread pool to be scheduled");
-                final String priority = (!savedJobPrioritiesStore.getJobSettings().getPutJobPriority().equals(resourceBundle.getString("defaultPolicyText"))) ? savedJobPrioritiesStore.getJobSettings().getPutJobPriority() : null;
-                //TODO There are two places we put jobs. This needs a refactor
-                final List<File> files = new ArrayList<>();
-                for (final File file : db.getFiles()) {
-                    if (file.isDirectory() && (file.listFiles().length == 0)) {
-                        final CreateFolderTask task = new CreateFolderTask(session.getClient(), bucket, file.getName(), loggingService, resourceBundle);
-                        workers.execute(task);
-                        task.setOnSucceeded(e -> {
-                            RefreshCompleteViewWorker.refreshCompleteTreeTableView(ds3Common, workers, loggingService);
-                            loggingService.logMessage("Created folder " + file.getName(), LogType.INFO);
-                        });
-                        task.setOnFailed(e -> {
-                            loggingService.logMessage("Could not create folder " + file.getName(), LogType.ERROR);
-                        });
-                    } else {
-                        files.add(file);
-                    }
-                    if (files.isEmpty()) {
-                        Ds3PanelService.refresh(selectedItem);
-                        return;
-                    }
-                    final Ds3PutJob putJob = new Ds3PutJob(session.getClient(), files, bucket, targetDir, priority,
-                            settingsStore.getProcessSettings().getMaximumNumberOfParallelThreads(), jobInterruptionStore,
-                            deepStorageBrowserPresenter, session, settingsStore, loggingService, resourceBundle);
-                    jobWorkers.execute(putJob);
-                    putJob.setOnSucceeded(e -> {
-                        LOG.info("Succeed");
-                        Ds3PanelService.refresh(selectedItem);
-                        ds3TreeTable.getSelectionModel().clearSelection();
-                        ds3TreeTable.getSelectionModel().select(selectedItem);
-                    });
-                    putJob.setOnFailed(e -> {
-                        LOG.info("setOnFailed");
-                        Ds3PanelService.refresh(selectedItem);
-                        ds3TreeTable.getSelectionModel().clearSelection();
-                        ds3TreeTable.getSelectionModel().select(selectedItem);
+            final List<Pair<String,String>> selectedItems = (List<Pair<String,String>>) db.getContent(DataFormat.lookupMimeType("local"));
+            final List<Pair<String, Path>> listofStringPath =  selectedItems.stream().map(i -> new Pair<String,Path>(i.getKey(), Paths.get(i.getValue()))).collect(Collectors.toList());
```

Comment:
This should use the `Guava.immutableList()` collector which will return an `ImmutableList` rather than a `List`